### DISCUSSION TITLE
TVM0.9环境，如何导入relax包

### DISCUSSION THREAD

**Discussion by qinsiyuan-cool** [mlc-ai/mlc-zh#150] [Q&A] [answered]
我没有下载mlc的环境，而是直接下载使用TVM 0.9的环境，于是在跑课程代码的时候，报“NameError: name 'relax' is not defined”错误，于是发现0.9版本的TVM还没有relax，然后我去了https://github.com/tlc-pack/relax/tree/relax/python/tvm ，把“relax”文件夹下载了下来，放到了服务器的”TVM/python/tvm/“文件夹下面再重新跑课程代码，仍然报错（下方图片），所以不清楚是我导入relax包出现了问题，还是暂时只能用mlc课程的环境跑课程代码，0.9版本的TVM还不能合并realx包？
![9b26dc612c15b190ff1c961f3ef8678](https://user-images.githubusercontent.com/82042694/190635220-f07a149d-3ab6-4c56-a1c5-642dab62ae8a.jpg)
![830b60d9a592c118d82a81b1c6bcc63](https://user-images.githubusercontent.com/82042694/190635245-c32eb660-925d-436d-83e4-290269f43b2e.jpg)

**Comment by yzh119** [comment, marked as answer]
Relax尚未进入TVM主分支，你需要安装mlc的环境。
导入Relax的python部分并足以让你在tvm主分支上使用Relax，Relax的大部分内容在C++上开发，Python端的代码只是Python接口。
建议直接使用mlc环境。

**Reply by vfdff** [reply]
这是因为 Relax 被集成到 mlc-ai 吗 ？
